Shared/Supporting Files/Views/WebView.swift

```diff
@@ -41,19 +41,18 @@ struct WebView: UIViewRepresentable {
     }
     
     class Coordinator: NSObject, WKNavigationDelegate {
-        func webView(
-            _ webView: WKWebView,
-            decidePolicyFor navigationAction: WKNavigationAction,
-            decisionHandler: @escaping (WKNavigationActionPolicy) -> Void
-        ) {
+        func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction) async -> WKNavigationActionPolicy {
             switch navigationAction.navigationType {
             case .linkActivated:
-                if let url = navigationAction.request.url, UIApplication.shared.canOpenURL(url) {
-                    UIApplication.shared.open(url)
+                if let url = navigationAction.request.url,
+                   await UIApplication.shared.canOpenURL(url) {
+                    Task { @MainActor in
+                        UIApplication.shared.open(url)
```

Comment:
I didn't find a way to suppress the warning for complaining that the `open` method takes in a non-Sendable dictionary, which literally is an empty dictionary by default. Thus I created its own little main actor context.

Reply:
`WKNavigationDelegate` is annotated as `MainActor`. So I think that you can try 2 things here instead:
1. Try annotating `func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction) async -> WKNavigationActionPolicy {` as MainActor
2. Use `MainActor.assumeIsolated` instead of kicking off a Task

Reply:
Annotating the delegate as MainActor causes warning, 

<img width="1281" alt="Screenshot 2024-07-10 at 8 24 51 AM" src="https://github.com/Esri/arcgis-maps-sdk-swift-samples/assets/4164028/2c302412-e534-4a33-b017-2cc6e5cea11f">

To fix, it will have to add `extension WKNavigationAction: @unchecked Sendable {}`

Then it works with this,

```
    class Coordinator: NSObject, WKNavigationDelegate {
        @MainActor
        func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction) async -> WKNavigationActionPolicy {
            switch navigationAction.navigationType {
            case .linkActivated:
                if let url = navigationAction.request.url,
                    UIApplication.shared.canOpenURL(url) {
                    await UIApplication.shared.open(url)
                }
                return .cancel
            default:
                return .allow
            }
        }
    }
```

Reply:
> To fix, it will have to add extension WKNavigationAction: @unchecked Sendable {}

We shouldn't do that.

There is no warning in Xcode 16 building with this code. The warning is only in Xcode 15.4.

```swift
    class Coordinator: NSObject, WKNavigationDelegate {
        @MainActor
        func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction) async -> WKNavigationActionPolicy {
            switch navigationAction.navigationType {
            case .linkActivated:
                if let url = navigationAction.request.url, UIApplication.shared.canOpenURL(url) {
                    await UIApplication.shared.open(url)
                }
                return .cancel
            default:
                return .allow
            }
        }
    }
```

I believe the reason is that WKNavigationAction is annotated as MainActor in the latest SDK:

```swift
@MainActor open class WKNavigationAction : NSObject {
```

Reply:
I do not see the warning with Complete Strict Concurrency with Xcode 16 Beta 3. But you can also try it before merging if you'd like.

Reply:
(Yes, I had that setting `strict concurrency = complete` when I tried it)

Reply:
Should we be fixing this based on Xcode 16 (still in beta) or Xcode 15.4?

Reply:
I've tested with Xcode 16b3 and it indeed doesn't give warning. I have the same question as Nimesh though - should I use the Xcode 16 compatible fix that still gives warning in Xcode 15, or should I keep it as-is?

Reply:
I would fix for Xcode 16. There is not a lot of point in fixing the warning now, if you have to re-fix it later.

Reply:
Whatever solution you go with, make sure that it works—without warning—on the minimum Xcode version (Xcode 15.0).

Reply:
> make sure that it works—without warning—on the minimum Xcode version (Xcode 15.0)

OK, I think my strategy will be turning on complete strict concurrency after `(the min req becomes Xcode 16) && (most if not all strict concurrency warnings are addressed)`. That way, those Xcode 16+ concurrency fixes won't cause trouble in any Xcode 15.0 min req releases.